test/OpenTelemetry.Tests/OpenTelemetry.Tests.csproj

```diff
@@ -16,7 +16,8 @@
   </ItemGroup>
 
   <ItemGroup>
-	<PackageReference Include="Microsoft.Extensions.Logging" />
+    <PackageReference Include="Microsoft.Extensions.Logging" />
+    <PackageReference Include="Microsoft.CSharp" Condition="'$(TargetFramework)' == 'net462'" />
```

Comment:
Why is this needed?

Reply:
These tests were being skipped on `NETFRAMEWORK`: https://github.com/open-telemetry/opentelemetry-dotnet/blob/104cf32d97b2aba7319635fc23d84d5f7663fe9b/test/OpenTelemetry.Tests/Logs/LogRecordTest.cs#L17

I found that odd because we support logging on `net462`+.

Turns out it is because of this one test using `dynamic`: https://github.com/open-telemetry/opentelemetry-dotnet/blob/104cf32d97b2aba7319635fc23d84d5f7663fe9b/test/OpenTelemetry.Tests/Logs/LogRecordTest.cs#L240

Adding `Microsoft.CSharp` for `net462` brings in `dynamic` support.